File: app/src/main/java/utils/DateHelper.java
package utils;

import android.annotation.SuppressLint;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;

@SuppressLint("SimpleDateFormat")
public class DateHelper {

    private final Date date;

    public DateHelper(long timeMillisecond) {
        Calendar calendar = Calendar.getInstance();
        calendar.setTimeInMillis(timeMillisecond);
        this.date = calendar.getTime();
    }

    public SimpleDateFormat yearFormatter() {
        return new SimpleDateFormat("yyyy");
    }

    public SimpleDateFormat monthFormatter() {
        return new SimpleDateFormat("MM");
    }

    public SimpleDateFormat dayFormatter() {
        return new SimpleDateFormat("dd");
    }

    public SimpleDateFormat hourFormatter() {
        return new SimpleDateFormat("HH");
    }

    public SimpleDateFormat minuteFormatter() {
        return new SimpleDateFormat("mm");
    }

    public int getYear() {
        int year = 0;
        try {
            year = Integer.parseInt(yearFormatter().format(date));
        } catch (Exception ignored) {
        }

        return year;
    }

    public int getMonth() {
        int month = 0;
        try {
            month = Integer.parseInt(monthFormatter().format(date));
        } catch (Exception ignored) {
        }

        return month;
    }

    public int getDay() {
        int day = 0;
        try {
            day = Integer.parseInt(dayFormatter().format(date));
        } catch (Exception ignored) {
        }

        return day;
    }

    public int getHour() {
        int hour = 0;
        try {
            hour = Integer.parseInt(hourFormatter().format(date));
        } catch (Exception ignored) {
        }

        return hour;
    }

    public int getMinute() {
        int minute = 0;
        try {
            minute = Integer.parseInt(minuteFormatter().format(date));
        } catch (Exception ignored) {
        }

        return minute;
    }

    public Date getDate() {
        return date;
    }

    public String getHourString() {
        return normalizeTime(getHour());
    }

    public String getMinuteString() {
        return normalizeTime(getMinute());
    }

    public String normalizeTime(int clock) {
        return clock < 10 ? "0" + clock : String.valueOf(clock);
    }

    public String getClock() {
        return getHourString() + ":" + getMinuteString();
    }

}

File: app/src/main/java/utils/DateHelper.kt
package utils

import android.annotation.SuppressLint
import java.text.SimpleDateFormat
import java.util.*

@SuppressLint("SimpleDateFormat")
class DateHelper(timeMillisecond: Long) {

    val date: Date

    private fun yearFormatter(): SimpleDateFormat = SimpleDateFormat("yyyy")
    private fun monthFormatter(): SimpleDateFormat = SimpleDateFormat("MM")
    private fun dayFormatter(): SimpleDateFormat = SimpleDateFormat("dd")

    private fun hourFormatter(): SimpleDateFormat = SimpleDateFormat("HH")
    private fun minuteFormatter(): SimpleDateFormat = SimpleDateFormat("mm")

    init {
        val calendar = Calendar.getInstance()
        calendar.timeInMillis = timeMillisecond
        date = calendar.time
    }

    fun getYear(): Int {
        var year = 0
        try {
            year = yearFormatter().format(date).toInt()
        } catch (ignored: Exception) {
        }
        return year
    }

    fun getMonth(): Int {
        var month = 0
        try {
            month = monthFormatter().format(date).toInt()
        } catch (ignored: Exception) {
        }
        return month
    }

    fun getDay(): Int {
        var day = 0
        try {
            day = dayFormatter().format(date).toInt()
        } catch (ignored: Exception) {
        }
        return day
    }

    fun getHour(): Int {
        var hour = 0
        try {
            hour = hourFormatter().format(date).toInt()
        } catch (ignored: Exception) {
        }
        return hour
    }

    fun getMinute(): Int {
        var minute = 0
        try {
            minute = minuteFormatter().format(date).toInt()
        } catch (ignored: Exception) {
        }
        return minute
    }

    val hourString: String
        get() = normalizeTime(getHour())

    val minuteString: String
        get() = normalizeTime(getMinute())

    private fun normalizeTime(clock: Int): String = if (clock < 10) "0$clock" else clock.toString()

    fun getClock(): String = "$hourString:$minuteString"

}